pkg/services/sso/redhatsso_service.go

```diff
@@ -0,0 +1,136 @@
+}
+
+func (r *redhatssoService) ListServiceAcc(accessToken string, ctx context.Context, first int, max int) ([]api.ServiceAccount, *errors.ServiceError) {
+	serviceAccounts, err := r.client.GetServiceAccounts(accessToken, first, max)
+	if err != nil {
+		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "failed to collect service accounts")
+	}
+
+	var saList []api.ServiceAccount
+	for _, sa := range serviceAccounts {
+		saList = append(saList, *convertServiceAccountDataToAPIServiceAccount(&sa))
+	}
+	return saList, nil
+}
+func (r *redhatssoService) RegisterKasFleetshardOperatorServiceAccount(accessToken string, agentClusterId string) (*api.ServiceAccount, *errors.ServiceError) {
+	return nil, nil
+}
+func (r *redhatssoService) DeRegisterKasFleetshardOperatorServiceAccount(accessToken string, agentClusterId string) *errors.ServiceError {
+	return nil
+}
+func (r *redhatssoService) GetServiceAccountById(accessToken string, ctx context.Context, id string) (*api.ServiceAccount, *errors.ServiceError) {
+	return r.GetServiceAccountByClientId(accessToken, ctx, id)
+}
+func (r *redhatssoService) GetServiceAccountByClientId(accessToken string, ctx context.Context, clientId string) (*api.ServiceAccount, *errors.ServiceError) {
+	serviceAccount, err := r.client.GetServiceAccount(accessToken, clientId)
+	if err != nil {
+		return nil, errors.NewWithCause(errors.ErrorServiceAccountNotFound, err, "service account not found clientId %s", clientId)
+	}
+	return convertServiceAccountDataToAPIServiceAccount(&serviceAccount), nil
+}
+func (r *redhatssoService) RegisterConnectorFleetshardOperatorServiceAccount(accessToken string, agentClusterId string) (*api.ServiceAccount, *errors.ServiceError) {
+	return nil, nil
+}
+func (r *redhatssoService) DeRegisterConnectorFleetshardOperatorServiceAccount(accessToken string, agentClusterId string) *errors.ServiceError {
+	return nil
+}
+func (r *redhatssoService) GetKafkaClientSecret(accessToken string, clientId string) (string, *errors.ServiceError) {
+	serviceAccount, err := r.client.GetServiceAccount(accessToken, clientId)
+	if err != nil {
+		return "", errors.NewWithCause(errors.ErrorFailedToGetSSOClientSecret, err, "failed to get sso client secret")
+	}
+
+	return *serviceAccount.Secret, nil
+}
+func (r *redhatssoService) CreateServiceAccountInternal(accessToken string, request services.CompleteServiceAccountRequest) (*api.ServiceAccount, *errors.ServiceError) {
+	return nil, nil
+}
+func (r *redhatssoService) DeleteServiceAccountInternal(accessToken string, clientId string) *errors.ServiceError {
+	return nil
+}
+
+//// utility functions
+func convertServiceAccountDataToAPIServiceAccount(data *serviceaccountsclient.ServiceAccountData) *api.ServiceAccount {
+	return &api.ServiceAccount{
+		ID:           *data.Id,
+		ClientID:     *data.ClientId,
+		ClientSecret: *data.Secret,
+		Name:         *data.Name,
+		Owner:        *data.OwnerId,
```

Comment:
@akoserwal  With the new API we don't have the owner, we just have the ownerId. Is it ok to return the OwnerId as Owner?

Reply:
The `owner` is deprecated going to be deprecated. I need to check with sso team if they have published new openapi spec for `createdBy`